File: src/net/dinkla/raytracer/lights/Ambient.java
package net.dinkla.raytracer.lights;

import net.dinkla.raytracer.colors.Color;
import net.dinkla.raytracer.hits.Shade;
import net.dinkla.raytracer.math.Ray;
import net.dinkla.raytracer.math.Vector3D;
import net.dinkla.raytracer.worlds.World;

public class Ambient extends Light {

    // emissive material
    public double ls;
    public Color color;

    public Ambient() {
        ls = 1.0;
        color = Color.WHITE;
    }
    
    @Override
    public Color L(World world, Shade sr) {
        return color.mult(ls);
    }

    @Override
    public Vector3D getDirection(Shade sr) {
        return Vector3D.Companion.getZERO();
    }

    @Override
    public boolean inShadow(World world, Ray ray, Shade sr) {
        return false; 
    }

}

File: src/main/kotlin/net/dinkla/raytracer/lights/Ambient.kt
package net.dinkla.raytracer.lights

import net.dinkla.raytracer.colors.Color
import net.dinkla.raytracer.hits.Shade
import net.dinkla.raytracer.math.Ray
import net.dinkla.raytracer.math.Vector3D
import net.dinkla.raytracer.worlds.World

// emissive material
open class Ambient(var ls: Double = 1.0, var color: Color = Color.WHITE) : Light() {

    override fun L(world: World, sr: Shade): Color = color * ls

    override fun getDirection(sr: Shade) = Vector3D.ZERO

    override fun inShadow(world: World, ray: Ray, sr: Shade): Boolean = false

}
